test/test_ops_gradients.py

```diff
@@ -10,8 +10,9 @@
 from torch.testing._internal.common_device_type import \
     (instantiate_device_type_tests, ops, OpDTypes)
 
-# TODO: fixme https://github.com/pytorch/pytorch/issues/68972
-torch.set_default_dtype(torch.float32)
```

Comment:
Not sure about this change? We actually WANT to keep this as `float32` as it is the default!

Reply:
(I agree that changing the default is not what we want ideally but just wanted to confirm. Sadly ScalarXScalar is still broken so can't get green signal from CI but changing the default dtype to `double` works locally)

The following script fails with this PR (based on failed test from `test_ops_gradient.py`)

```python
import torch

from torch.testing._internal.common_methods_invocations import op_db

# torch.set_default_dtype(torch.double) # Fails if this is not set.

op = list(filter(lambda op: (op.name == 'H'), op_db))[0]

samples = op.sample_inputs('cpu', torch.complex128, requires_grad=True)

for sample in samples:
    torch.autograd.gradgradcheck(op.op, sample.input, check_fwd_over_rev=True)
```

Haven't verified yet, but maybe the `gradcheck/gradgradcheck` code splits complex128 tensor into `real` (double tensor) and `imag` (double tensor) part does computation and tries to combine them back.

For default dtype of `torch.float`,
Before this PR, `1j * torch.double -> torch.complex128`, everything passed. (but wrong type-promotion).
After this PR, `1j * torch.double -> torch.complex64`, tests start failing. (correct type-promotion).

Does this explain the probable issue @albanD?

Reply:
> After this PR, 1j * torch.double -> torch.complex64, tests start failing. (correct type-promotion).

That is the right type promotion? That would be loosing precision for the double arg no?

Reply:
a complex scalar multiplied with a cdouble tensor should produce a cdouble tensor, though. The type promotion for float x complex is different from long x float.

Reply:
Thanks for the clarity! I think I understand it better now. Apologies for the noise :)!